errorlint/lint.go

```diff
@@ -189,6 +189,27 @@ func LintErrorComparisons(fset *token.FileSet, info *TypesInfoExt) []Lint {
 			continue
 		}
 
+		hasErrorCase := false
```

Comment:
Could you move this into a function like `isSwitchOnNil`? That helps explain what this block does.